internal/mode/static/state/graph/graph.go

```diff
@@ -183,6 +187,17 @@ func BuildGraph(
 
 	npCfg := getNginxProxy(state.NginxProxies, processedGwClasses.Winner)
 	gc := buildGatewayClass(processedGwClasses.Winner, npCfg, state.CRDMetadata, validators.GenericValidator)
+	if gc != nil && npCfg != nil && npCfg.Source != nil {
+		spec := npCfg.Source.Spec
+		globalSettings = &policies.GlobalPolicySettings{
+			NginxProxyValid:  npCfg.Valid,
+			TelemetryEnabled: npCfg.Valid && spec.Telemetry != nil && spec.Telemetry.Exporter != nil,
+		}
+
+		if spec.Telemetry != nil {
+			globalSettings.TracingSpanAttributes = spec.Telemetry.SpanAttributes
```

Comment:
why isn't guarded by check if npCfg is valid?

Reply:
Doesn't really matter in the end. When we validate the policy, if npCfg is invalid, we'll mark the policy as invalid, and we won't generate the code for it.